src/lib.rs

```diff
@@ -374,11 +476,15 @@ where
     I: Interface,
 {
     /// Initializes the BME280
-    fn init<D: DelayUs>(&mut self, delay: &mut D) -> Result<(), Error<I::Error>> {
+    fn init<D: DelayUs>(
+        &mut self,
+        delay: &mut D,
+        config: Configuration,
+    ) -> Result<(), Error<I::Error>> {
```

Comment:
Do you mind leaving the current 'init' function as is that will default to the current configuration and add new function. Perhaps 'init_with_config' that will take in the configuration struct?